go-to/code.ts

```diff
@@ -0,0 +1,84 @@
+const types = [
+  "BOOLEAN_OPERATION",
+  "COMPONENT",
+  "COMPONENT_SET",
+  "CONNECTOR",
+  "DOCUMENT",
+  "ELLIPSE",
+  "FRAME",
+  "GROUP",
+  "INSTANCE",
+  "LINE",
+  "PAGE",
+  "POLYGON",
+  "RECTANGLE",
+  "SHAPE_WITH_TEXT",
+  "SLICE",
+  "STAMP",
+  "STAR",
+  "STICKY",
+  "TEXT",
+  "VECTOR",
+];
+
+// The 'input' event listens for text change in the Quick Actions box after a plugin is 'Tabbed' into.
+figma.parameters.on('input', ({ key, query, result }) => {
+  switch (key) {
+      case 'name':
+          const filter = node => node.name.toLowerCase().startsWith(query.toLowerCase());
+
+          // Always suggest all of the pages in the file 
+          const pages = figma.root
+            .findChildren(node => query.length === 0 ? true : filter(node))
+            .map(node => ({ name: `${node.name} (page)`, data: { name: node.name, id: node.id } }));
+
+          // Only show layers when the user types a query 
+          const nodes = query.length > 0 ? figma.currentPage
+              .findAll(node => filter(node)) : [];
+
+          const formattedNodes = nodes.map((node) => {
+              const name = `${node.name} [${node.id}]`;
+              return ({ name, data: { name: node.name, id: node.id } });
+          });
+          const suggestions = [...pages, ...formattedNodes];
+          result.setSuggestions(suggestions);
+          break;
+      default:
+          return;
+  }
+});
+
+// When the user presses Enter after inputting all parameters, the 'run' event is fired.
+figma.on('run', ({ parameters }) => {
+  if (parameters) {
```

Comment:
`parameters` should always be true here, though probably not in the typings. For the purposes of the sample, it could be clearer to just do something like: `startPluginWithParameters(parameters!);`